### ISSUE TITLE
JavadocParagraph: define in documentation list of validated block-tags

### ISSUE THREAD

**Issue by bradleylarrick** [checkstyle/checkstyle#15865]
https://checkstyle.org/checks/javadoc/javadocparagraph.html#Description

Warnings are generated about the presence (or lack of presence) of \<p> tag in javadoc text that is inconsistent with google-java-format-1.24.0.

```
$ cat Test.java 
package org.natuna.test;

/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <div style="padding: 1em 1em 1em">
 * <table>
 * <caption></caption>
 * <thead>
 * <tr>
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
 * </tr>
 * </thead>
 * <tbody>
 * <tr>
 * <td>East</td> <td>E</td>
 * </tr>
 * </tbody>
 * </table>
 * </div>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Default constructor. */
  public Test() {
    super();
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml Test.java 
Starting audit...
Audit done.

$ java -jar google-java-format-1.24.0-all-deps.jar Test.java > TestUpdated.java
$ cat TestUpdated.java 
package org.natuna.test;

/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <p><div style="padding: 1em 1em 1em">
 *
 * <table>
 * <caption></caption>
 * <thead>
 * <tr>
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
 * </tr>
 * </thead>
 * <tbody>
 * <tr>
 * <td>East</td> <td>E</td>
 * </tr>
 * </tbody>
 * </table>
 *
 * </div>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Default constructor. */
  public Test() {
    super();
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml TestUpdated.java 
Starting audit...
[WARN] /var/tmp/TestUpdated.java:6:4: <p> tag should not precede HTML block-tag
      '<div>', <p> tag should be removed. [JavadocParagraph]
[WARN] /var/tmp/TestUpdated.java:27:1: The name of the outer type and the file do not match. [OuterTypeFilename]
Audit done.
```

Expected: no violation `tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]`
Violation from OuterTypeFilename is side effect for formatter output to new file.

**Comment by romani** [comment]
Please share output of CLI for both tools. It helps a lot to approve issue and let someone do a fix

**Comment by bradleylarrick** [comment]
````
[~/temp] cat -n Test.java
     1  /*
     2   * Copyright (c) 2023-2024 Bradley Larrick. All rights reserved.
     3   *
     4   * Licensed under the Apache License v2.0
     5   * http://www.apache.org/licenses/LICENSE-2.0
     6   */
     7
     8  package org.natuna.test;
     9
    10  /**
    11   * Plugin to standardize all directions to their appropriate abbreviation.
    12   *
    13   * <div style="padding: 1em 1em 1em">
    14   * <table>
    15   * <caption></caption>
    16   * <thead>
    17   * <tr>
    18   * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
    19   * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
    20   * </tr>
    21   * </thead>
    22   * <tbody>
    23   * <tr>
    24   * <td>East</td> <td>E</td>
    25   * </tr>
    26   * </tbody>
    27   * </table>
    28   * </div>
    29   *
    30   * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
    31   */
    32  public class Test {
    33
    34    /** Default constructor. */
    35    public Test() {
    36      super();
    37    }
    38  }
[~/temp] java -jar java -jar google-java-format-1.24.0-all-deps.jar Test.java > Test.new
[~/temp] diff -u Test.java Test.new
--- Test.java   2024-11-05 13:39:51.280283100 -0500
+++ Test.new    2024-11-05 13:40:18.894847900 -0500
@@ -10,7 +10,8 @@
 /**
  * Plugin to standardize all directions to their appropriate abbreviation.
  *
- * <div style="padding: 1em 1em 1em">
+ * <p><div style="padding: 1em 1em 1em">
+ *
  * <table>
  * <caption></caption>
  * <thead>
@@ -25,6 +26,7 @@
  * </tr>
  * </tbody>
  * </table>
+ *
  * </div>
  *
  * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
[~/temp] cp Test.new Test.java
[~/temp] java -jar java -jar checkstyle-10.20.0-all.jar -c google_checks.xml Test.java
Starting audit...
[WARN] C:\Users\bradley\temp\Test.java:13:4: <p> tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]
Audit done.
[~/temp]
````

**Comment by romani** [comment]
@bradleylarrick, thanks a lot.
Do you have any open source project that we add to our regression testing to make sure we will check all violations before releasing a fix?

**Comment by romani** [comment]
@bradleylarrick ,

> `-c google_checks.xml`

do you use some modification of google_checks.xml ? if not better to do `-c /google_checks.xml` it uses config from jar

**Comment by romani** [comment]
@Zopsss , please help to fix.

**Comment by bradleylarrick** [comment]
Your suggestion doesn't work in my environment. Specifying the checks file with the preceding slash causes an error.
````
[~/temp] ls -l
total 45433
-rw-r--r-- 1 bradley 197614      646 Oct 29 15:36 FooBar.java
-rw-r--r-- 1 bradley 197614      622 Oct 29 18:19 FooBarNew.java
-rw-r--r-- 1 bradley 197614     4683 Oct 29 09:13 LocalStyle.xml
-rw-r--r-- 1 bradley 197614      860 Nov  5 13:40 Test.java
-rw-r--r-- 1 bradley 197614      860 Nov  5 13:40 Test.new
-rw-r--r-- 1 bradley 197614      560 Nov  5 13:37 Test.sav
-rw-r--r-- 1 bradley 197614 19630203 Oct 29 15:12 checkstyle-10.19.0-all.jar
-rw-r--r-- 1 bradley 197614 19630557 Nov  4 07:45 checkstyle-10.20.0-all.jar
-rw-r--r-- 1 bradley 197614  3613813 Nov  5 10:22 google-java-format-1.23.0-all-deps.jar
-rw-r--r-- 1 bradley 197614  3614987 Oct 29 16:36 google-java-format-1.24.0-all-deps.jar
[~/temp] java -jar checkstyle-10.20.0-all.jar -c google_checks.xml Test.java
Starting audit...
[WARN] C:\Users\bradley\temp\Test.java:13:4: <p> tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]
Audit done.
[~/temp] java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml Test.java
Could not find config XML file 'C:/Program Files/Git/google_checks.xml'.
````
The two methods work the same in a DOS window, but I'm using the GNU bash shell that comes with git.

**Comment by romani** [comment]
uhh, this is interesting windows effect.
it supposed to search in filesystem and if not found search in jar resources.

**Comment by rnveach** [comment]
This isn't a Windows issue, but it is with mingw (or whatever interpreter your using on Windows). It's modifying the input with a '/' to the fully qualified file path and thus fails as the file doesn't exist. '\' or no slash works as the input is passed as the name with no slash.

**Comment by romani** [comment]
@bradleylarrick, please try
`-c "/google_checks.xml"`.

**Comment by romani** [comment]
There might be some special case with AST on such HTML,
We have test on P tag right before some other tag 
https://github.com/checkstyle/checkstyle/blob/3dc2a3d713b21f005a12219803f8a5f08d0a8dd1/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect4.java#L54-L58

**Comment by romani** [comment]
```
$ java -jar checkstyle-10.20.0-all.jar -j javadoc1.javadoc 
JAVADOC -> JAVADOC [0:0]
|--TEXT -> Plugin to standardize all directions to their appropriate abbreviation. [0:0]
|--NEWLINE -> \n [0:71]
|--HTML_ELEMENT -> HTML_ELEMENT [1:0]
|   `--P_TAG_START -> P_TAG_START [1:0]
|       |--START -> < [1:0]
|       |--P_HTML_TAG_NAME -> p [1:1]
|       `--END -> > [1:2]
|--HTML_ELEMENT -> HTML_ELEMENT [1:3]
|   `--HTML_TAG -> HTML_TAG [1:3]
|       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [1:3]
|       |   |--START -> < [1:3]
|       |   |--HTML_TAG_NAME -> div [1:4]
|       |   |--WS ->   [1:7]
|       |   |--ATTRIBUTE -> ATTRIBUTE [1:8]
|       |   |   |--HTML_TAG_NAME -> style [1:8]
|       |   |   |--EQUALS -> = [1:13]
|       |   |   `--ATTR_VALUE -> "padding: 1em 1em 1em" [1:14]
|       |   `--END -> > [1:37]
|       |--NEWLINE -> \n [1:38]
|       |--HTML_ELEMENT -> HTML_ELEMENT [2:0]
|       |   `--HTML_TAG -> HTML_TAG [2:0]
|       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [2:0]
|       |       |   |--START -> < [2:0]
|       |       |   |--HTML_TAG_NAME -> table [2:1]
|       |       |   `--END -> > [2:6]
|       |       |--NEWLINE -> \n [2:7]
|       |       |--HTML_ELEMENT -> HTML_ELEMENT [3:0]
|       |       |   `--HTML_TAG -> HTML_TAG [3:0]
|       |       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [3:0]
|       |       |       |   |--START -> < [3:0]
|       |       |       |   |--HTML_TAG_NAME -> caption [3:1]
|       |       |       |   `--END -> > [3:8]
|       |       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [3:9]
|       |       |           |--START -> < [3:9]
|       |       |           |--SLASH -> / [3:10]
|       |       |           |--HTML_TAG_NAME -> caption [3:11]
|       |       |           `--END -> > [3:18]
|       |       |--NEWLINE -> \n [3:19]
|       |       |--HTML_ELEMENT -> HTML_ELEMENT [4:0]
|       |       |   `--THEAD -> THEAD [4:0]
|       |       |       |--THEAD_TAG_START -> THEAD_TAG_START [4:0]
|       |       |       |   |--START -> < [4:0]
|       |       |       |   |--THEAD_HTML_TAG_NAME -> thead [4:1]
|       |       |       |   `--END -> > [4:6]
|       |       |       |--NEWLINE -> \n [4:7]
|       |       |       |--TR -> TR [5:0]
|       |       |       |   |--TR_TAG_START -> TR_TAG_START [5:0]
|       |       |       |   |   |--START -> < [5:0]
|       |       |       |   |   |--TR_HTML_TAG_NAME -> tr [5:1]
|       |       |       |   |   `--END -> > [5:3]
|       |       |       |   |--NEWLINE -> \n [5:4]
|       |       |       |   |--TH_TAG_START -> TH_TAG_START [6:0]
|       |       |       |   |   |--START -> < [6:0]
|       |       |       |   |   |--TH_HTML_TAG_NAME -> th [6:1]
|       |       |       |   |   |--WS ->   [6:3]
|       |       |       |   |   |--ATTRIBUTE -> ATTRIBUTE [6:4]
|       |       |       |   |   |   |--HTML_TAG_NAME -> scope [6:4]
|       |       |       |   |   |   |--EQUALS -> = [6:9]
|       |       |       |   |   |   `--ATTR_VALUE -> "col" [6:10]
|       |       |       |   |   |--WS ->   [6:16]
|       |       |       |   |   |--ATTRIBUTE -> ATTRIBUTE [6:17]
|       |       |       |   |   |   |--HTML_TAG_NAME -> style [6:17]
|       |       |       |   |   |   |--EQUALS -> = [6:22]
|       |       |       |   |   |   `--ATTR_VALUE -> "padding: .3em 4em 0 0; text-align: left" [6:23]
|       |       |       |   |   `--END -> > [6:65]
|       |       |       |   |--TEXT -> Direction [6:66]
|       |       |       |   |--NEWLINE -> \n [6:75]
|       |       |       |   |--TH_TAG_START -> TH_TAG_START [7:0]
|       |       |       |   |   |--START -> < [7:0]
|       |       |       |   |   |--TH_HTML_TAG_NAME -> th [7:1]
|       |       |       |   |   |--WS ->   [7:3]
|       |       |       |   |   |--ATTRIBUTE -> ATTRIBUTE [7:4]
|       |       |       |   |   |   |--HTML_TAG_NAME -> scope [7:4]
|       |       |       |   |   |   |--EQUALS -> = [7:9]
|       |       |       |   |   |   `--ATTR_VALUE -> "col" [7:10]
|       |       |       |   |   |--WS ->   [7:16]
|       |       |       |   |   |--ATTRIBUTE -> ATTRIBUTE [7:17]
|       |       |       |   |   |   |--HTML_TAG_NAME -> style [7:17]
|       |       |       |   |   |   |--EQUALS -> = [7:22]
|       |       |       |   |   |   `--ATTR_VALUE -> "padding: .3em 4em 0 0; text-align: left" [7:23]
|       |       |       |   |   `--END -> > [7:65]
|       |       |       |   |--TEXT -> Abbreviation [7:66]
|       |       |       |   |--NEWLINE -> \n [7:78]
|       |       |       |   `--TR_TAG_END -> TR_TAG_END [8:0]
|       |       |       |       |--START -> < [8:0]
|       |       |       |       |--SLASH -> / [8:1]
|       |       |       |       |--TR_HTML_TAG_NAME -> tr [8:2]
|       |       |       |       `--END -> > [8:4]
|       |       |       |--NEWLINE -> \n [8:5]
|       |       |       `--THEAD_TAG_END -> THEAD_TAG_END [9:0]
|       |       |           |--START -> < [9:0]
|       |       |           |--SLASH -> / [9:1]
|       |       |           |--THEAD_HTML_TAG_NAME -> thead [9:2]
|       |       |           `--END -> > [9:7]
|       |       |--NEWLINE -> \n [9:8]
|       |       |--HTML_ELEMENT -> HTML_ELEMENT [10:0]
|       |       |   `--TBODY -> TBODY [10:0]
|       |       |       |--TBODY_TAG_START -> TBODY_TAG_START [10:0]
|       |       |       |   |--START -> < [10:0]
|       |       |       |   |--TBODY_HTML_TAG_NAME -> tbody [10:1]
|       |       |       |   `--END -> > [10:6]
|       |       |       |--NEWLINE -> \n [10:7]
|       |       |       |--TR -> TR [11:0]
|       |       |       |   |--TR_TAG_START -> TR_TAG_START [11:0]
|       |       |       |   |   |--START -> < [11:0]
|       |       |       |   |   |--TR_HTML_TAG_NAME -> tr [11:1]
|       |       |       |   |   `--END -> > [11:3]
|       |       |       |   |--NEWLINE -> \n [11:4]
|       |       |       |   |--TD -> TD [12:0]
|       |       |       |   |   |--TD_TAG_START -> TD_TAG_START [12:0]
|       |       |       |   |   |   |--START -> < [12:0]
|       |       |       |   |   |   |--TD_HTML_TAG_NAME -> td [12:1]
|       |       |       |   |   |   `--END -> > [12:3]
|       |       |       |   |   |--TEXT -> East [12:4]
|       |       |       |   |   `--TD_TAG_END -> TD_TAG_END [12:8]
|       |       |       |   |       |--START -> < [12:8]
|       |       |       |   |       |--SLASH -> / [12:9]
|       |       |       |   |       |--TD_HTML_TAG_NAME -> td [12:10]
|       |       |       |   |       `--END -> > [12:12]
|       |       |       |   |--TEXT ->   [12:13]
|       |       |       |   |--TD -> TD [12:14]
|       |       |       |   |   |--TD_TAG_START -> TD_TAG_START [12:14]
|       |       |       |   |   |   |--START -> < [12:14]
|       |       |       |   |   |   |--TD_HTML_TAG_NAME -> td [12:15]
|       |       |       |   |   |   `--END -> > [12:17]
|       |       |       |   |   |--TEXT -> E [12:18]
|       |       |       |   |   `--TD_TAG_END -> TD_TAG_END [12:19]
|       |       |       |   |       |--START -> < [12:19]
|       |       |       |   |       |--SLASH -> / [12:20]
|       |       |       |   |       |--TD_HTML_TAG_NAME -> td [12:21]
|       |       |       |   |       `--END -> > [12:23]
|       |       |       |   |--NEWLINE -> \n [12:24]
|       |       |       |   `--TR_TAG_END -> TR_TAG_END [13:0]
|       |       |       |       |--START -> < [13:0]
|       |       |       |       |--SLASH -> / [13:1]
|       |       |       |       |--TR_HTML_TAG_NAME -> tr [13:2]
|       |       |       |       `--END -> > [13:4]
|       |       |       |--NEWLINE -> \n [13:5]
|       |       |       `--TBODY_TAG_END -> TBODY_TAG_END [14:0]
|       |       |           |--START -> < [14:0]
|       |       |           |--SLASH -> / [14:1]
|       |       |           |--TBODY_HTML_TAG_NAME -> tbody [14:2]
|       |       |           `--END -> > [14:7]
|       |       |--NEWLINE -> \n [14:8]
|       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [15:0]
|       |           |--START -> < [15:0]
|       |           |--SLASH -> / [15:1]
|       |           |--HTML_TAG_NAME -> table [15:2]
|       |           `--END -> > [15:7]
|       |--NEWLINE -> \n [15:8]
|       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [16:0]
|           |--START -> < [16:0]
|           |--SLASH -> / [16:1]
|           |--HTML_TAG_NAME -> div [16:2]
|           `--END -> > [16:5]
|--NEWLINE -> \n [16:6]
|--HTML_ELEMENT -> HTML_ELEMENT [17:0]
|   `--P_TAG_START -> P_TAG_START [17:0]
|       |--START -> < [17:0]
|       |--P_HTML_TAG_NAME -> p [17:1]
|       `--END -> > [17:2]
|--TEXT -> This plugin assumes numbered streets have the correct suffix, when applicable. [17:3]
|--NEWLINE -> \n [17:81]
`--EOF -> <EOF> [18:0]

```

**Comment by Zopsss** [comment]
> Expected: no violation `tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]`
Violation from OuterTypeFilename is side effect for formatter output to new file.

The violation is expected, `<div>` is a block-level tag and google style guide says that a block level tag should not be preceded with `<p>`.

https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs

> HTML tags for other block-level elements, such as `<ul>` or `<table>`, are not preceded with `<p>`.

Your code: 
```
 * <p><div style="padding: 1em 1em 1em">
```

The `<p>` is added by the formatter so it seems that it is formatter's fault. The `<div>` is a block-level tag and formatter adds a preceding `<p>` tag before it which breaks style guide rule.

This rule is also mentioned in the Check's [description](https://checkstyle.org/checks/javadoc/javadocparagraph.html#Description):

> First paragraph tag should not precede [HTML block-tag](https://www.w3schools.com/html/html_blocks.asp), nested paragraph tags are allowed to do that.


Here's an example from the Check's documentation which might help you to better understand Check's working

```java
// violation 4 lines below '<p> tag should not precede HTML block-tag '<h1>''
/**
 * No tag (ok).
 *
 * <p><h1>P tag is preceding block-tag h1, not allowed</h1>
 *
 * <p><b>P tag is preceding an inline-tag b, this is allowed</b>
 */
```

`<b>` is an inline tag so there'll be no violation for it but `<h1>` is a block-level tag so it'll give violation. 

I hope this answers your question 🙂

@bradleylarrick

**Comment by bradleylarrick** [comment]
@Zopsss, are you going to report this to the google-java-format team, or shall I?

**Comment by Zopsss** [comment]
> @Zopsss, are you going to report this to the google-java-format team, or shall I?

I will do it once I get on my laptop

**Comment by romani** [comment]
Question to all:
Do you think that we need to mention all tag names in our doc ?

>Here are the block-level elements in HTML:
~<address><article><aside><blockquote><canvas><dd><div><dl><dt><fieldset><figcaption><figure><footer><form><h1>-<h6><header><hr><li><main><nav><noscript><ol><p><pre><section><table><tfoot><ul><video>`

So users will clearly see them. 
Link that we have is not direct, so users might be confused what checkstyle consider as block tags.
One more reason is such list at link might be updated with new tag , but checkstyle will continue to validate old list.

**Comment by romani** [comment]
@bradleylarrick, in future don't hesitate to open issue on formatter.
Our team is not participating in such project.

**Comment by Zopsss** [comment]
> Question to all: Do you think that we need to mention all tag names in our doc ?
> 
> > Here are the block-level elements in HTML:
> > `<address><article><aside><blockquote><canvas><dd><div><dl><dt><fieldset><figcaption><figure><footer><form><h1>-<h6><header><hr><li><main><nav><noscript><ol><p>\<pre\><section><table><tfoot><ul><video>`
> 
> So users will clearly see them. Link that we have is not direct, so users might be confused what checkstyle consider as block tags. One more reason is such list at link might be updated with new tag , but checkstyle will continue to validate old list.

+1, this is a great idea. 

> One more reason is such list at link might be updated with new tag , but checkstyle will continue to validate old list.

This didn't came to my mind when I wrote the doc, thanks for pointing this out!!!

**Comment by romani** [comment]
```
$ cat Test.java 
public class Test {
    /**
     * Plugin to standardize all directions
     *
     * <p><dt>some text</dt> some text out
     *
     * <p>This plugin assumes numbered streets
     */
    public void plugin() {}
}

$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.3+9-LTS
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:5: warning: empty <p> tag
     * <p><dt>some text</dt> some text out
       ^
Test.java:5: error: tag not allowed here: <dt>
     * <p><dt>some text</dt> some text out
          ^
Test.java:1: warning: no comment
public class Test {
       ^
Test.java:1: warning: use of default constructor, which does not provide a comment
public class Test {
       ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./overview-tree.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./index-all.html...
Generating ./search.html...
Generating ./index.html...
Generating ./help-doc.html...
1 error
3 warnings
```

so not all HTML bock-tags are supported by javadoc tool, we have to define exact list of tags.

**Comment by Zopsss** [comment]
Some guy on Stackoverflow has posted which HTML tags are supported by Java: https://stackoverflow.com/a/30957355/18727281

I tried finding an official list of HTML tags supported by javadoc so we can filter them, but it seems like there isn't a list available like that, the most I was able to find was: https://docs.oracle.com/en/java/javase/17/docs/specs/javadoc/doc-comment-spec.html#block-tags

In the above mentioned Stackoverflow answer, the user suggested using `Xdoclint` to validate Javadoc:

> Javadoc's doclint component enforces this restriction. You can disable all doclint warnings by passing -Xdoclint:none to javadoc, though you should consider fixing your Javadoc comments because otherwise the generated HTML API documentation may look bad or may omit content. (I usually use -Xdoclint:all,-missing to get warnings about everything except missing Javadoc @ tags.)

maybe this can be useful for us?

**Comment by Zopsss** [comment]
@romani issue opened on GJF's side: https://github.com/google/google-java-format/issues/1180

@bradleylarrick If you're using Checkstyle and GJF both but if GJF is incorrectly formats javadoc, you can use their `--skip-javadoc-formatting` option to skip javadoc formatting.

```java
/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <div>Some javadoc here....</div>
 *
 * <ul>
 * <li>Item - 1 </li>
 * <li>Item - 2 </li>
 * <li>Item - 3 </li>
 * </ul>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Some javadoc. */
  public static void main(String[] args) {

  }
}
```

```
$ java -jar .\google-java-format-1.24.0-all-deps.jar --skip-javadoc-formatting .\Test.java > .\TestUpdated.java
```

```java
/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <div>Some javadoc here....</div>
 *
 * <ul>
 * <li>Item - 1 </li>
 * <li>Item - 2 </li>
 * <li>Item - 3 </li>
 * </ul>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Some javadoc. */
  public static void main(String[] args) {}
}
```

But this also seems to have some inconsistencies discussed at: https://github.com/google/google-java-format/issues/789

**Comment by romani** [comment]
>most I was able to find was: https://docs.oracle.com/en/java/javase/17/docs/specs/javadoc/doc-comment-spec.html#block-tags

This is about javadoc block tags, but in this issue we discuss html block tags

**Comment by Zopsss** [comment]
Yeah, my bad. I didn't noticed it
